codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -395,14 +386,11 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
             verifyPrefaceReceived();
 
             Http2Stream stream = connection.requireStream(streamId);
-            verifyRstStreamNotReceived(stream);
             if (stream.state() == CLOSED) {
                 // RstStream frames must be ignored for closed streams.
                 return;
             }
 
-            stream.resetReceived();
-
             listener.onRstStreamRead(ctx, streamId, errorCode);
```

Comment:
@nmittler - IIRC the `InboundHttp2ToHttpAdapter` is capturing this to see if there is an existing message, and if so just fire up the pipeline what we have received (because it is now "complete").  If by `handling general stream closure` you mean the `Http2Stream.Listener` interface then: yes `InboundHttp2ToHttpAdapter` is already a listener but I'm not sure we can achieve the same result (because there is no `ctx`).  So it looks like we may end up dropping the messages (from the clients point of view) if we don't have this call.